Composer/packages/client/src/ShellApi.ts

```diff
@@ -132,8 +132,14 @@ export const ShellApi: React.FC = () => {
     if (id === undefined) throw new Error('must have a file id');
     const file = lgFiles.find(file => file.id === id);
     if (!file) throw new Error(`lg file ${id} not found`);
-
-    return lgUtil.parse(file.content);
+    return lgUtil.parse(file.content).map(template => {
+      const { name, parameters, body } = template;
+      return {
+        name,
+        parameters,
+        body,
+      };
+    });
```

Comment:
This is not clear to me why this would make any difference?

Reply:
Raw parse result contains complicate class `LGTemplate` not only just `{ name, parameters, body }` but also has some prototype methods cannot be cloned and send by browser post message api. And data consumer may don't need it, so here filter them.

Reply:
By the way, `templates` data is already provided by `getData` api, maybe there do not have a very strong reason to add this `getLgTemplates`  method in shellApi. @yeze322

Reply:
Good catch @zhixzhan , please confirm this, if not necessary, we can just remove it.

Reply:
@a-b-r-o-w-n https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm here is the reference zhixiang sent to me about copy behavior in postMessage